v2/googlecloud-to-elasticsearch/docs/BigQueryToElasticsearch/README.md

```diff
@@ -0,0 +1,256 @@
+# BigQuery to Elasticsearch Dataflow Template
+
+The [BigQueryToElasticsearch](../../src/main/java/com/google/cloud/teleport/v2/elasticsearch/templates/BigQueryToElasticsearch.java) pipeline ingests
+data from a BigQuery table into Elasticsearch. The template can either read the entire table or read using a supplied query.
+
+Pipeline flow is illustrated below:
+
+![alt text](../img/bigquery-to-elasticsearch-dataflow.png "BigQuery to Elasticsearch pipeline flow")
+
+## Getting Started
+
+### Requirements
+* Java 8
+* Maven
+* BigQuery table exists
+* Cloud Storage bucket exists
+
+### Building Template
+This is a Flex Template meaning that the pipeline code will be containerized, and the container will be
+run on Dataflow.
+
+#### Building Container Image
+* Set environment variables that will be used in the build process.
+```sh
+export PROJECT=<my-project>
+export IMAGE_NAME=<my-image-name>
+export BUCKET_NAME=gs://<bucket-name>
+export TARGET_GCR_IMAGE=gcr.io/${PROJECT}/${IMAGE_NAME}
+export BASE_CONTAINER_IMAGE=gcr.io/dataflow-templates-base/java8-template-launcher-base
+export BASE_CONTAINER_IMAGE_VERSION=latest
+export TEMPLATE_MODULE=bigquery-to-elasticsearch
+export APP_ROOT=/template/${TEMPLATE_MODULE}
+export COMMAND_SPEC=${APP_ROOT}/resources/${TEMPLATE_MODULE}-command-spec.json
+export TEMPLATE_IMAGE_SPEC=${BUCKET_NAME}/images/${TEMPLATE_MODULE}-image-spec.json
+
+export INPUT_TABLE_SPEC=<my-project:my-dataset.my-table>
+export NODE_ADDRESSES=<comma-separated-list-nodes>
+export INDEX=<my-index>
+export DOCUMENT_TYPE=<my-type>
+export USE_LEGACY_SQL=false
+
+gcloud config set project ${PROJECT}
+```
+* Build and push image to Google Container Repository
+```sh
+mvn clean package \
+    -Dimage=${TARGET_GCR_IMAGE} \
+    -Dbase-container-image=${BASE_CONTAINER_IMAGE} \
+    -Dbase-container-image.version=${BASE_CONTAINER_IMAGE_VERSION} \
+    -Dapp-root=${APP_ROOT} \
+    -Dcommand-spec=${COMMAND_SPEC} \
+    -am -pl ${TEMPLATE_MODULE}
+```
+
+#### Creating Image Spec
+
+Create file in Cloud Storage with path to container image in Google Container Repository.
+```sh
+echo '{
+    "image":"'${TARGET_GCR_IMAGE}'",
+    "metadata":{"name":"Replicates BigQuery to Elasticsearch",
+    "description":"Replicates BigQuery data into an Elasticsearch index",
+    "parameters":[
```

Comment:
Parameters should match the latest changes as per
https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/master/v2/elasticsearch-common/src/main/java/com/google/cloud/teleport/v2/elasticsearch/options/ElasticsearchWriteOptions.java

Same comments are applicable for all Readme.md files